src/Integration.Vsix/CFamily/CMake/CompilationDatabaseRequestFactory.cs

```diff
@@ -47,7 +47,7 @@ public IRequest TryGet(string analyzedFilePath, CFamilyAnalyzerOptions analyzerO
             }
 
             // TODO - handle user specifying the language via a command / argument #2533
-            var languageKey = CFamilyShared.FindLanguageFromExtension(analyzedFilePath);
+            var languageKey = CFamilyShared.FindLanguageFromExtension(dbEntry.File);
```

Comment:
We need to take the extension from the code entry, not the header file. Added an explicit test for this.